corehq/apps/app_manager/fields.py

```diff
@@ -436,7 +436,8 @@ def _get_modules_and_forms(self, as_dict=True):
 
             app_langs.append(form['app']['langs'][0] if 'langs' in form['app'] else 'en')
             app_id = form['app']['id'] if has_app else self.UNKNOWN_SOURCE
-            module = form.get('module')
+            # Set module to None if it doesn't have an ID (FB 285678)
+            module = form['module'] if 'module' in form and 'id' in form['module'] else None
```

Comment:
actually can just modify the following line to be. Should not make the module disappear if it does not have the id?
```
module_id = (module['id'] 
                       if has_app and module is not None and 'id' in module
                       else self.UNKNOWN_SOURCE)
```

Reply:
Yes. But will that cause a problem on the next line? If it doesn't have an 'id' attribute, does it have a 'name' attribute?

Reply:
The PR title said if it does not have an `id` so i thought that was the only thing missing and being handled. If it does not have a 'name', is it even a module?

Reply:
Indeed! What's worse is that this error is inconsistent. Come back later and suddenly form['module']['id'] has a value! :roll_eyes: 

That makes me think the source of the problem might be a service, and not code.

I guess the code needs to  handle gracefully whatever it gets. (But it would be nice no know exactly what it's getting.)

Reply:
what do you mean by "come back later"?

Reply:
I mean that if you follow the repro steps in the project spaces mentioned in [FB 285678][1] or [FB 285774][2] they no longer result in an error. So something other than the code has changed.

We know that `form['module']['id']` has a value because in one of the project spaces all forms have modules. Or in other words, there are no forms for which `form['module']` is `None`.


  [1]: https://manage.dimagi.com/default.asp?285678
  [2]: https://manage.dimagi.com/default.asp?285774